web/pdf_page_view.js

```diff
@@ -1007,6 +998,12 @@ class PDFPageView {
     canvasWrapper.append(canvas);
     this.canvas = canvas;
 
+    // Adding the loading class is slightly postponed in order to not have
+    // it with loadingIcon.
+    // If we don't do that the visibility of the background is changed but the
+    // transition isn't triggered.
+    setTimeout(() => canvasWrapper.classList.add("loading"), 0);
+
```

Comment:
Shouldn't this be placed after e.g. [this line instead](https://github.com/mozilla/pdf.js/pull/15992/files#diff-c48e3561004f5db8f11d5ebab2fd661591222ba911cb4173fbced15f026bac6bR801), to make the loadingIcon work with both canvas and svg mode?